content/Contributing to open source: An overview of the django-drip-campaings project - Bruno Michetti.md

```diff
@@ -0,0 +1,217 @@
+
+#### Message class
+
+By default, Django Drip creates and sends messages that are instances of Django’s `EmailMultiAlternatives` class. If you want to customize in any way the message that is created and sent, you can do that by creating a subclass of `EmailMessage` and overriding any method that you want to behave differently. For example:
+
+```python
+from django.core.mail import EmailMessage
+from drip.drips import DripMessage
+
+class PlainDripEmail(DripMessage):
+
+    @property
+    def message(self):
+        if not self._message:
+            email = EmailMessage(self.subject, self.plain, self.from_email, [self.user.email])
+            self._message = email
+        return self._message
+```
+
+In that example, `PlainDripEmail` overrides the message property of the base `DripMessage` class to create a simple `EmailMessage` instance instead of an `EmailMultiAlternatives` instance.
+
+In order to be able to specify that your custom message class should be used for a drip, you need to configure it in the `DRIP_MESSAGE_CLASSES` setting:
+
+```python
+DRIP_MESSAGE_CLASSES = {
+    'plain': 'myproj.email.PlainDripEmail',
+}
+
+```
+
+This will allow you to choose in the admin, for each drip, whether the `default` (`Dripmessage`) or `plain` message class should be used for generating and sending the messages to users.
+
+#### Send drips
+
+To send the created and enabled drips, run the command:
+
+```
+python manage.py send_drips
+```
+
+You can use cron to schedule the drips.
+
+### What we have been doing in the project
+
+As I said, we didn't create this project. We forked it and started to make some fixes and improvements. The first thing we did was to upgrade the used Django version. Now it has django 3.0.7.  
+Besides we started to create the documentation with [Sphinx](https://www.sphinx-doc.org/en/master/) to have this project in [Read the Docs](https://readthedocs.org/) in the future.  
+Another enhancement was to add [flake8](https://flake8.pycqa.org/en/latest/) to enforce the use of style guides in the code. Finally we added an integration with [travis-ci](https://travis-ci.com/) to have a continuous integration service on any change in the project.  
+We registered and fixed some issues such as the view of the User fields, and configuration problems to run the app locally. There are some issues to address and discover yet so if you like OSS you are more than welcome to work with Django Drip Campaigns.  
+In the open source initiative we wanted to publish this app as a pypi package, so I'll explain this a little more in the next section.
+
+### Publish a pypi package
+
+Although there is work to do with the Django Drip Campaigns project, we already published the package in pypi. It's not something very hard to do, but it's important to know a few things before starting, so in this section I'm going to talk about the main points of publishing the app as a package.
```

Comment:
`Despite Django Drip Campaigns is still a work in progress, we...` in place of `Although...`, sounds more smooth

Reply:
`...the main steps for publishing the app...`

Reply:
Lets try this:

```
Despite Django Drip Campaigns is still a work in progress, we already published the package in PyPI. It's nothing fancy, but it's a must to know a few things before starting, so in this section, I'll be talking about the main steps for publishing an app.
```